arkouda/index.py

```diff
@@ -113,6 +116,96 @@ def lookup(self, key):
 
         return in1d(self.index, key)
 
+    def save(self, prefix_path: str, dataset: str = 'index', mode: str = 'truncate',
+             compressed: bool = False, file_format: str = 'HDF5') -> str:
+        """
+        Save the index to HDF5 or Parquet. The result is a collection of files,
+        one file per locale of the arkouda server, where each filename starts
+        with prefix_path. Each locale saves its chunk of the array to its
+        corresponding file.
+
+        Parameters
+        ----------
+        prefix_path : str
+            Directory and filename prefix that all output files share
+        dataset : str
+            Name of the dataset to create in files (must not already exist)
+        mode : str {'truncate' | 'append'}
+            By default, truncate (overwrite) output files, if they exist.
+            If 'append', attempt to create new dataset in existing files.
+        compressed : bool
+            Defaults to False. When True, files will be written with Snappy compression
+            and RLE bit packing. This is currently only supported on Parquet files and will
+            not impact the generated files when writing HDF5 files.
+        file_format : str {'HDF5', 'Parquet'}
+            By default, saved files will be written to the HDF5 file format. If
+            'Parquet', the files will be written to the Parquet file format. This
+            is case insensitive.
+
+        Returns
+        -------
+        string message indicating result of save operation
+
+        Raises
+        ------
+        RuntimeError
+            Raised if a server-side error is thrown saving the pdarray
```

Comment:
`saving the Index`?

Reply:
@pierce314159 - I left it as pdarray because it is only saving self.index which is a pdarray. It made sense in my head, but I am happy to update if you think it should be changed.